libdnf5/module/module_sack.cpp

```diff
@@ -739,6 +739,24 @@ ModuleSack::resolve_active_module_items() {
 }
 
 
+ModuleQuery module_spec_to_query(BaseWeakPtr & base, const std::string & module_spec) {
+    auto nsvcaps = Nsvcap::parse(module_spec);
+
+    ModuleQuery query(base, true);
+    for (auto & nsvcap : nsvcaps) {
+        ModuleQuery nsvcap_query(base, false);
+        nsvcap_query.filter_nsvca(nsvcap);
+        query |= nsvcap_query;
```

Comment:
May I ask you why you would like to use all possibilities as a candidate? For RPM we use the first valid candidate form that match. 
Practically I think there will be not real difference in result, because module spec parsing is more exact.

Reply:
Oh, right, I didn't realize it's sufficient to take the first candidate only. I changed it.